test/unit/cms/page_test.rb

```diff
@@ -97,4 +97,48 @@ def setup
     page.expects(:tag_list).returns([]).once
     page.as_json(include: [:tag_list])
   end
+
+  test 'generates a system_name with proper format from title when creating' do
+    page = CMS::Page.new.tap do |p|
+      p.title = 'New Page'
+      p.provider = @provider
+      p.section = @provider.sections.root
+      p.path = '/new-page'
+    end
+
+    page.save!
+
+    assert_equal 'new-page', page.reload.system_name
+  end
+
+  test 'generates a system_name with proper format from title when updating' do
+    page = CMS::Page.new.tap do |p|
+      p.title = 'New Page'
+      p.system_name = 'sysname-1'
+      p.provider = @provider
+      p.section = @provider.sections.root
+      p.path = '/new-page'
+    end
+    page.save!
+
+    page.system_name = ''
+    page.save!
+
+    assert_equal 'new-page', page.reload.system_name
+  end
+
+  test "doesn't generate a system_name when one is given" do
```

Comment:
What about if given an invalid system name, such as `'sysname 1'`?

Reply:
It will return a validation error because the `system_name` format for pages is enforced in the parent class `CMS::Template` https://github.com/3scale/porta/blob/1ece547ea9568f4ed9c845ea477dfaa7b1e451c7/app/models/cms/template.rb#L19-L20

Reply:
Yes, I mean what about writing a test for that scenario.

Reply:
Done